cmd/rendmp/src/lib.rs

```diff
@@ -920,6 +937,93 @@ fn rendmp_blackbox(
     Ok(())
 }
 
+fn rendmp_open_pin(
+    subargs: RendmpArgs,
+    hubris: &HubrisArchive,
+    core: &mut dyn humility::core::Core,
+    context: &mut HiffyContext,
+) -> Result<()> {
+    let (addr, dev) = check_addr(&subargs, hubris)?;
+    let op = hubris.get_idol_command("Power.rendmp_dma_read")?;
+    let regs: &'static [u16] = match dev {
+        ISL_DEV_NAME => &[
+            0x00BD, // open-pin
+            0xE925, // mask
+        ],
+        RAA_DEV_NAME => &[
+            0x00BE, 0x00BF, // open-pin
+            0xE904, 0xE905, // mask
+        ],
+        _ => panic!("wrong device ({dev})"),
+    };
+    let mut ops = vec![];
+    for r in regs {
+        let payload = op.payload(&[
+            ("addr", IdolArgument::Scalar(addr as u64)),
+            ("reg", IdolArgument::Scalar(*r as u64)),
+        ])?;
+        context.idol_call_ops(&op, &payload, &mut ops)?;
+    }
+    ops.push(hif::Op::Done);
+    let results = context.run(core, ops.as_slice(), None)?;
+
+    // Decode Hiffy results
+    let mut values = vec![];
+    for r in results {
+        match hiffy_decode(hubris, &op, r)? {
+            Err(e) => bail!("hiffy error: {e}"),
+            Ok(Value::Base(Base::U32(b))) => values.push(b),
+            v => bail!("unexpected type in result: {v:?}"),
+        }
+    }
+
+    // Register decoding was determined with a mix of Power Navigator
+    // experiments and discussion with Renesas.
+    let (open, mask, phases) = match dev {
+        ISL_DEV_NAME => {
+            let (open, mask) = (values[0] as u64, values[1] as u64);
+            let mut phases = vec![];
+            for phase in 0..6 {
+                phases.push((phase.to_string(), phase * 2 + 6));
+            }
+            for i in 0..3 {
+                phases.push((format!("VSEN{i}"), i * 2 + 24));
+            }
+            (open, mask, phases)
+        }
+        RAA_DEV_NAME => {
+            let open = values[0] as u64 | (values[1] as u64) << 32;
+            let mask = values[2] as u64 | (values[3] as u64) << 32;
+            let mut phases = vec![];
+            for phase in 0..20 {
+                phases.push((phase.to_string(), phase * 2));
+            }
+            for i in 0..2 {
+                phases.push((format!("VSEN{i}"), i * 2 + 40));
+            }
+            (open, mask, phases)
+        }
+        _ => unreachable!(), // checked above
+    };
+    println!("Open pin check for {} at {addr:#x}", dev.to_uppercase());
+    println!(" {} | {}", "PHASE".bold(), "STATE".bold());
+    println!("-------|-------");
+    for (name, bit) in phases {
+        let is_open = (open & mask & (0b11 << bit)) != 0;
```

Comment:
Even tinier nit - `(0b11 << bit)` caught me off guard because that's masking two bits, not one bit. I'm not sure there's a better name here though; `bit_offset` is maybe not any clearer? Feel free to ignore this.

Reply:
Renamed to `bit_offset` and (more importantly) added some comments